torch_ipex/csrc/aten_ipex_bridge.cpp

```diff
@@ -496,24 +495,52 @@ std::vector<at::Tensor> shallowFallbackToCPUTensorList(const at::TensorList& ten
   return dpcpp_tensor_vec;
 }
 
-void cvtTensorToScalaraType(const at::Tensor& ipexTensor, at::ScalarType dstScalarType) {
+
+void reorderTensorToScalarTypeForDNNL(const at::Tensor& ipexTensor, at::ScalarType dstScalarType) {
+  if (ipexTensor.device().type() == at::DeviceType::CPU) {
+    return reorderTensorToScalaraType(ipexTensor, dstScalarType);
```

Comment:
this branch handles the conversion before OP returns?
if so, can use last branch (w/o shade) to handle this?

Reply:
The API is used before the dispatch to DNNL op. reorderTensorToScalarTypeForDNNL => Dispatch to DNNL.

Reply:
if this API is called at the beginning of OPs, Is there any chance the device will be CPU?

Reply:
Yes. "to" should be the case. CPU tensor converts to IPEX tensor.

Reply:
Sorry, misunderstood your comment point. I will fix it.